### PR TITLE
Revise package loader + introduce tests

### PR DESCRIPTION
**Pull request by mikejolley**
Moves the autoloader/package loader logic to new classes that only load with PHP 5.6+

This also removes API unit tests, and introduces some tests to ensure packages are built.

Need to see how travis handles this before merge.

_This series of PRs target the [Feature Plugin Package](https://github.com/woocommerce/woocommerce/pull/23957) branch where all related functionality will be grouped, and stack on top of one another. If reviewed, please mark as approved to merge, but leave the merging until last since other PRs will target this one._

### CODE REVIEW

File: packages/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# External Packages
```
Comment (timmyc) [inline packages/README.md]: Thanks for the great documentation here Mike, and the example repo! 👏

File: src/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# WooCommerce `src` files
+
+This directory is home to new WooCommerce class files under the \Automattic\WooCommerce\ namespace using PSR-4 file naming. This is to take full advantage of autoloading.
+
+Currently, these classes have a PHP 5.6 requirement. No required core classes will be added here until this PHP version is enforced. If running an older version of PHP, these class files will not be used.
+
+## Installing Composer
+
+Composer is used to generate autoload class-maps for the files here. The stable release of WooCommerce comes with the autoloader, however, if you're running a development version you'll need to use Composer.
```
Comment (timmyc) [inline src/README.md]: Guessing we should update the general README and CONTRIBUTING documentation to add in similar instructions

File: packages/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+Finally, you will need to tell core to load your package. Edit `src/Packages.php` and add your package to the list of packages there:
+
+```php
+	protected static $packages = [
+		'woocommerce-blocks'          => '\\Automattic\\WooCommerce\\Blocks\\Package',
+    'woocommerce-rest-api'        => '\\Automattic\\WooCommerce\\RestApi\\Package',
+    'woocommerce-example-package' => '\\Automattic\\WooCommerce\\ExamplePackage\\Package',
+	];
+```
+
```
Comment (timmyc) [inline packages/README.md]: Might be nice to also prompt adding similar test cases as you have done in `tests/unit-tests/packages/packages.php` when adding in a new package too.

File: woocommerce.php [hunk 4]
```diff
@@ -19,47 +19,26 @@
 }
 
 /**
- * Load all the packages.
+ * Load core packages and the autoloader.
  *
- * We want to fail gracefully if `composer install` has not been executed yet, so we are checking for the autoloader.
- * If the autoloader is not present, let's log the failure and display a nice admin notice.
+ * The new packages and autoloader require PHP 5.6+. If this dependency is not met, do not include them. Users will be warned
```
Comment (timmyc) [inline woocommerce.php]: So does this make a breaking change in 3.7 for users on < 5.6? Especially if they use the REST API or blocks currently?

**Review by timmyc — COMMENTED** [review]
One question about if this is breaking or not in 3.7 when running on < 5.6 - and just a few other minor notes. The documentation here is fantastic!

File: src/README.md (hunk 2 above)
Comment (mikejolley) [reply src/README.md]: Yes, I'll add this to the todo list on main PR

File: woocommerce.php (hunk 4 above)
Comment (mikejolley) [reply woocommerce.php]: @timmyc yes - I brought this up in Proton today. They are happy with this min-requirement, we just need to look at how core enforces this outside of this PR. If core does it, the version check code in this branch can go.

**Pull request merged**